lib/phase/flatten.js

```diff
@@ -2,9 +2,32 @@
 
 const Hoek = require('hoek');
 const clone = require('clone');
+const TEMPLATE_REGEX = require('screwdriver-data-schema').config.regex.FULL_TEMPLATE_NAME;
+const NAME_MATCH = 1;
+const VERSION_MATCH = 3;
+const LABEL_MATCH = 7;
 
 /**
- * Overlays the stuff specified in a specific job on top of the defaults in shared
+ * Merge oldJob into newJob
+ * "oldJob" takes precedence over "newJob". For ex: individual job settings vs shared settings
+ * @param  {Object}   newJob    Job to be merged into. For ex: shared settings
+ * @param  {Object}   oldJob    Job to merge. For ex: individual job settings
+ */
+function merge(newJob, oldJob) {
+    // Intialize new job with default fields
+    newJob.environment = newJob.environment || {};
+    newJob.secrets = newJob.secrets || [];
+    newJob.settings = newJob.settings || {};
+
+    // Merge
+    Object.assign(newJob.environment, oldJob.environment || {});
+    Object.assign(newJob.settings, oldJob.settings || {});
```

Comment:
The default value for `.secrets` is an array: https://github.com/screwdriver-cd/config-parser/pull/32/files#diff-daa88a14d1ada8a3aa7d881f9424ba2cR19 Is this also supposed to be an array?